File: scanner/scanner/models/collection.py
from __future__ import annotations

from ..utils import Language, Model
from .genre import Genre
from .metadataid import MetadataId


class Collection(Model):
	slug: str
	original_language: Language | None
	genres: list[Genre]
	rating: int | None
	external_id: dict[str, MetadataId]

	translations: dict[Language, CollectionTranslation] = {}


class CollectionTranslation(Model):
	name: str
	latin_name: str | None
	description: str | None
	tagline: str | None
	aliases: list[str]
	tags: list[str]

	poster: str | None
	thumbnail: str | None
	banner: str | None
	logo: str | None

File: front/src/models/collection.ts
import { z } from "zod/v4";
import { Genre } from "./utils/genre";
import { KImage } from "./utils/images";
import { Metadata } from "./utils/metadata";
import { zdate } from "./utils/utils";

export const Collection = z
	.object({
		id: z.string(),
		slug: z.string(),
		name: z.string(),
		original: z.object({
			name: z.string(),
			latinName: z.string().nullable(),
			language: z.string(),
		}),
		tagline: z.string().nullable(),
		aliases: z.array(z.string()),
		tags: z.array(z.string()),
		description: z.string().nullable(),
		rating: z.number().int().gte(0).lte(100).nullable(),
		startAir: zdate().nullable(),
		endAir: zdate().nullable(),
		genres: z.array(Genre),
		externalId: Metadata,

		poster: KImage.nullable(),
		thumbnail: KImage.nullable(),
		banner: KImage.nullable(),
		logo: KImage.nullable(),

		createdAt: zdate(),
		updatedAt: zdate(),
	})
	.transform((x) => ({
		...x,
		href: `/collections/${x.slug}`,
	}));

export type Collection = z.infer<typeof Collection>;
